### DISCUSSION TITLE
Types for Exceptions

### DISCUSSION THREAD

**Discussion by fastsyrup** [Terreii/use-pouchdb#94] [Q&A]
Hi there,

I have the following code:
``` 
import { useDoc, usePouch } from "use-pouchdb";
...
try {
    const res = await db.get('some doc');
  } catch (error) {
    ...
  }
}
``` 
I get the following typescript error: 'error' is of type 'unknown'.ts(18046)

Where can I find the type of error potentially thrown?

Thanks!

**Comment by Terreii** [comment]
usePouchDB does use [PouchDB's error package](https://www.npmjs.com/package/pouchdb-errors). The types for errors are in [types/pouchdb-core](https://www.npmjs.com/package/@types/pouchdb-core).

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pouchdb-core/index.d.ts#L99-L110

**Reply by fastsyrup** [reply]
Thx! Sorry but after installing these as dev dependencies I still get the following typescript error for the following simple try catch: 
```
const db = usePouch<PageDataType>();
try {
    const res = await db.get(notebookName);
  } catch (error) {
...
}
```
Typescript says:
```
'error' is of type 'unknown'.
```
How can I get rid of the error, besides ignoring it?

**Reply by Terreii** [reply]
Could you try a `error as  as PouchDB.Core.Error` like in:
https://github.com/Terreii/use-pouchdb/blob/5fbc3960f3ad26c339e4be006b9383cb19f0cb38/src/useDoc.ts#L101
